src/main/java/io/fusionauth/http/server/internal/HTTPWorker.java

```diff
@@ -267,9 +268,27 @@ public void run() {
       logger.debug(String.format("[%s] Closing socket with status [%d]. An IO exception was thrown during processing. These are pretty common.", Thread.currentThread().threadId(), Status.InternalServerError), e);
       closeSocketOnError(response, Status.InternalServerError);
     } catch (Throwable e) {
-      // Log the error and signal a failure
+      HTTPUnexpectedExceptionHandler unexpectedExceptionHandler = configuration.getDefaultExceptionHandler();
       var status = Status.InternalServerError;
-      logger.error(String.format("[%s] Closing socket with status [%d]. An HTTP worker threw an exception while processing a request.", Thread.currentThread().threadId(), status), e);
+      if (unexpectedExceptionHandler != null) {
+        if (response != null) {
```

Comment:
Why isn't this code in the default handler?

Reply:
It could be - was waffling, I can create a default and move it.